learn/safe-core-account-abstraction-sdk/onramp-kit.md

```diff
@@ -1,3 +1,101 @@
+
+> Currently we are providing both the public key and the server for testing purposes. In the future you will need to use your own public key and server based on the final documentation Stripe will provide once their on ramps solution is ready for production. See the [considerations and limitations](#considerations-and-limitations) section for more details.
+
+Once the instance is created, you can call the `open(SafeOnRampOpenOptions)` method to start the session with the provider and opening the widget.
+
+As an example, you can use the following code:
+
+```typescript
+const sessionData = await safeOnRamp.open({
+  walletAddress,
+  networks: ['polygon']
+  element: '#stripe-root',
+  sessionId: 'cos_1Mei3cKSn9ArdBimJhkCt1XC', // Optional, if you want to use a specific created session
+  events: {
+    onLoaded: () => console.log('Loaded'),
+    onPaymentSuccessful: () => console.log('Payment successful')
+    onPaymentError: () => console.log('Payment failed')
+    onPaymentProcessing: () => console.log('Payment processing')
+  }
+})
+```
+
+### Considerations and limitations
+
+1. The library is under development and it's not ready for production use. We are working with Stripe and participating in the pilot testing for their new [on ramp solution](https://stripe.com/es/blog/crypto-onramp)
+
+    Considering this, we provide a public key and a deployed server only available for testing purposes. It can be used like this:
+
+```typescript
+const safeOnRamp = await SafeOnRampKit.init(SafeOnRampProviderType.Stripe, {
+  onRampProviderConfig: {
+    stripePublicKey:
+      'pk_test_51MZbmZKSn9ArdBimSyl5i8DqfcnlhyhJHD8bF2wKrGkpvNWyPvBAYtE211oHda0X3Ea1n4e9J9nh2JkpC7Sxm5a200Ug9ijfoO', // Safe public key
+    onRampBackendUrl: 'https://safe-onramp-backend.5afe.dev', // Safe deployed server
+  },
+});
+```
+
+2. As we are working on Stripe _testmode_, the purchases are simulated. You can use the fake data in the [docs](https://stripe.com/docs/testing) and the [following test cards](https://stripe.com/docs/testing?testing-method=card-numbers#cards) to enter the required information in the embedded widget.
+
+3. When using with testnets as Mumbai in Polygon, the crypto assets will be transferred so PLEASE DO TRY TO USE LOWER AMOUNTS to preserve testnets liquidity, but SPECIALLY WITH USDC ON POLYGON MUMBAI.
+
+4. If you want to deploy a POC with your solution, bear in mind that our integration with Stripe has the following domains whitelisted:
+
+   - localhost: For local development
+   - [netlify.app](https://www.netlify.com) and [vercel.app](https://vercel.com) for hosted deployments
+
+    So you can deploy your solution in one of these hosting providers. Using a different domain won't work and the widget will throw an error.
+
+5. Currently the Stripe widget can only be used if you are __based in the United States__. If you are hacking from another country you could use a VPN connecting to an US server.
```

Comment:
Are we allowed to tell people to use VPN in docs? @yagopv

Reply:
Not sure. I think we can suggest it but let's add @clemoon and @dasanra to the loop to get their feedback

Reply:
I had the same doubt, so maybe it's good that we remove that part.